crates/ruma-client/src/client/builder.rs

```diff
@@ -82,6 +82,7 @@ impl ClientBuilder {
                 )
                 .await?
                 .known_versions()
+                .into_iter()
                 .collect(),
```

Comment:
This is a bit unfortunate, maybe we should change the client to hold a `BTreeSet` too? Anyways, can be a separate PR.

Reply:
Well the issue is that `OutgoingRequest::try_into_http_request` needs a slice, so it's easier to store it as a `Vec`.